internal/server/iigointernal/orchestration.go

```diff
@@ -74,10 +74,25 @@ func RunIIGO(g *gamestate.GameState, clientMap *map[shared.ClientID]baseclient.C
 	iigoClients = *clientMap
 
 	// Increments the budget by a constant 100
-	// TODO:- the constant should be retrieved from the rules
-	g.IIGORolesBudget[shared.President] += 100
-	g.IIGORolesBudget[shared.Judge] += 100
-	g.IIGORolesBudget[shared.Speaker] += 100
+	PresidentIncRule := rules.AvailableRules["increment_budget_president"].ApplicableMatrix
+	PresidentIncrement := PresidentIncRule.At(0, 1)
+	g.IIGORolesBudget[shared.President] += shared.Resources(PresidentIncrement)
+	JudgeIncRule := rules.AvailableRules["increment_budget_judge"].ApplicableMatrix
+	JudgeIncrement := JudgeIncRule.At(0, 1)
+	g.IIGORolesBudget[shared.Judge] += shared.Resources(JudgeIncrement)
+	SpeakerIncRule := rules.AvailableRules["increment_budget_speaker"].ApplicableMatrix
+	SpeakerIncrement := SpeakerIncRule.At(0, 1)
+	g.IIGORolesBudget[shared.Speaker] += shared.Resources(SpeakerIncrement)
+
+	variablesToCache := []rules.VariableFieldName{rules.PresidentIncrement}
+	valuesToCache := [][]float64{{PresidentIncrement}}
+	monitoring.addToCache(g.PresidentID, variablesToCache, valuesToCache)
+	variablesToCache = []rules.VariableFieldName{rules.JudgeIncrement}
+	valuesToCache = [][]float64{{JudgeIncrement}}
+	monitoring.addToCache(g.PresidentID, variablesToCache, valuesToCache)
+	variablesToCache = []rules.VariableFieldName{rules.SpeakerIncrement}
+	valuesToCache = [][]float64{{SpeakerIncrement}}
+	monitoring.addToCache(g.PresidentID, variablesToCache, valuesToCache)
```

Comment:
I don't understand the need for this logging. By what the rule is it would almost always be failed, and here we would be putting the blame on the president? I think this is unnecessary.

Reply:
Functionally yes, it's unnecessary. I just put it in there for the parallelism with the other rules, in case the logging served another purpose other than to always rule-check (for instance, actual logging). Given the code structure, the rule would always succeed bc RoleIncrement is drawn from the rule and therefore always satisfying the rule.

Reply:
Ohhhhh I see, You are right, it would not fail. Yet, at the same time, I'm still queazy about logging it in case it fails :D For example, what if the rule is changed in IIGO?

Reply:
If it makes you iffy about possible glitches, all those lines can just be discarded - again, it was just for semantics and serves 0 functionality 😆